### PR TITLE
Handle deprecated signatures in Scala3doc

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

File: scala3doc/src/dotty/dokka/model/api/api.scala [hunk 1]
```diff
@@ -124,6 +126,7 @@ extension[T] (member: Member):
 
   def signature: Signature = memberExt.fold(Signature(name))(_.signature)
   def asLink: LinkToType = LinkToType(signature, dri, kind)
+  def deprecated: Option[Annotation] = memberExt.flatMap(_.annotations.find(a => a.dri.getPackageName == "scala" && a.dri.getClassNames == "deprecated"))
```
Comment (romanowski) [inline scala3doc/src/dotty/dokka/model/api/api.scala:129]: What if we are have a deprecated member from Java?

**Review by romanowski — COMMENTED** [review]
Minor code style issue plus question about Java.

If Java does not work we can add support for it in a followup PR

File: scala3doc/src/dotty/dokka/model/api/api.scala (hunk 1 above)
Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/model/api/api.scala:129]: It depends. It still will be recognized as an annotation, though it won't have added strikethrough style. The question is: if we really find that using `java.lang.Deprecated` in Scala code should cause any special treatment. In my opinion it's user fault

File: scala3doc/src/dotty/dokka/model/api/api.scala [hunk 2]
```diff
@@ -80,10 +80,12 @@ enum Origin:
 case class Annotation(val dri: DRI, val params: List[Annotation.AnnotationParameter])
 
 object Annotation:
-  sealed trait AnnotationParameter
-  case class PrimitiveParameter(val name: Option[String] = None, val value: String) extends AnnotationParameter
-  case class LinkParameter(val name: Option[String] = None, val dri: DRI, val value: String) extends AnnotationParameter
-  case class UnresolvedParameter(val name: Option[String] = None, val unresolvedText: String) extends AnnotationParameter
+  sealed trait AnnotationParameter {
```
[the remark this answers is not in the document]
Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/model/api/api.scala:83]: Will change

File: scala3doc/src/dotty/dokka/model/api/api.scala (hunk 1 above)
Comment (romanowski) [reply scala3doc/src/dotty/dokka/model/api/api.scala:129]: What if one want to use a Java class with deprecated members (e.g. [this one](https://docs.oracle.com/en/java/javase/13/docs/api/java.desktop/javax/swing/JComponent.AccessibleJComponent.html) )

Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/model/api/api.scala:129]: You mean as a base class for Scala class... Yeah, we would like to make the Inherited methods deprecated. Will add support then

Comment (romanowski) [reply scala3doc/src/dotty/dokka/model/api/api.scala:129]: I think this can be done in follow up PR

**Pull request merged**